packages/node/src/login/oidc/ClientRegistrar.ts

```diff
@@ -32,10 +32,40 @@ import {
   IClient,
   IClientRegistrarOptions,
   ConfigurationError,
+  determineSigningAlg,
+  PREFERRED_SIGNING_ALG,
 } from "@inrupt/solid-client-authn-core";
 import { Client, Issuer } from "openid-client";
 import { configToIssuerMetadata } from "./IssuerConfigFetcher";
 
+export function negotiateClientSigningAlg(
+  issuerConfig: IIssuerConfig,
+  clientPreference: string[]
+): string {
+  if (issuerConfig.idTokenSigningAlgValuesSupported === undefined) {
```

Comment:
I'm having a bit trouble navigating the codebase from the PR interview, but have you tested this using an `issuerConfig` in which this property is not declared, rather than undefined? i.e.

```
issuerConfig = {}
```

rather than

```
issuerConfig = { idTokenSigningAlgValuesSupported: undefined }
```

Off the top of my head I'm not sure whether the same holds for object properties as for variables, but if it does, the former might be problematic (which can be solved by doing a check for `typeof ... === "undefined"`).

Reply:
Actually, might be good to use  `Array.isArray()` anyway, just to also guard against some more defined-but-invalid values.

Reply:
a90e36b19f5ac153d425356eb01a746b5f52f1a5